### ISSUE TITLE
Connection header default value must be capitalized

### ISSUE THREAD

**Issue by mco-system** [aio-libs/aiohttp#5660]
According to the The WebSocket Protocol RFC (https://tools.ietf.org/html/rfc6455#section-11.2)

"The request MUST contain a |Connection| header field whose value  MUST include the "Upgrade" token."

In aiohttp/client.py, the default value for the Connection header is "upgrade" which must be capitalized as "Upgrade"

See: https://github.com/aio-libs/aiohttp/blob/4c663a2c887d8a18e5f4220288e3324569e34b8b/aiohttp/client.py#L711

This lowercase "Connection" header value prevent client connection to servers that implement strictly the RFC by forcing the server to immediately stop the handshake protocol (this is how I found this issue).

**Comment by mco-system** [comment]
It looks like it has been fixed in 4.x.x but still present in 3.7.x
https://github.com/aio-libs/aiohttp/blob/d487af9a62aac62b752cd4e412a9cf1c0c6b214d/aiohttp/hdrs.py#L89

**Comment by webknjaz** [comment]
The RFC section you linked only talks about the value (`WebSocket` token), not the header name. How HTTP headers work is governed by another RFC but FWIW they are case incentive.

**Comment by webknjaz** [comment]
You probably wanted to link https://tools.ietf.org/html/rfc6455#section-4.1 but it doesn't mention anything being case sensitive in the value. Moreover, some examples in the RFC show lowercase values.

**Comment by webknjaz** [comment]
So I looked closer into the RFCs and the older ones have `Connection: Upgrade` in their examples but the newer ones have `Connection: upgrade` and occasionally the titlecased version.
Ref: https://tools.ietf.org/html/rfc7230#section-6.7.

Also, I've noticed two spellings for the protocol token: `WebSocket` and `websocket`. They are listed on this page http://www.iana.org/assignments/http-upgrade-tokens/http-upgrade-tokens.xhtml with links to the respective RFC sections that introduced them.

Based on this, it sounds like it's the server in the question that does not follow RFC, not aiohttp.
